### ISSUE TITLE
ecp-data-vis-sdk on ARM: ispc@1.19.0: cmake fails: Either X86, ARM, WASM or XE targets need to be enabled.

### ISSUE THREAD

**Issue by eugeneswalker** [LLNL/spack#38044]
### Steps to reproduce the issue

For ARM, the following `ecp-data-vis-sdk` specs fail to build due to failure of dependency `ispc@1.19.0` to build
* `ecp-data-vis-sdk +adios2 +ascent +cinema ~cuda +darshan +faodel +hdf5 +paraview +pnetcdf ~rocm +sz +unifyfs +veloc +visit +vtkm +zfp`
* `ecp-data-vis-sdk +adios2 +hdf5 +vtkm +zfp +paraview +cuda cuda_arch=75`
* `ecp-data-vis-sdk +adios2 +hdf5 +vtkm +zfp +paraview +cuda cuda_arch=80`

Using
* `spack@develop` (9ee2d79de172de14477a78e5d407548a63eea33a from `Tue May 16 13:32:26 2023 -0500`)
* Ubuntu 20.04, ARM64
* GCC 11.1.0

<details><summary>Concretization:</summary>
<pre>
 -   ispc@1.19.0%gcc@11.1.0~ipo build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-aarch64
[+]      ^bison@3.8.2%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^diffutils@3.9%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]              ^libiconv@1.17%gcc@11.1.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-aarch64
[+]          ^gnuconfig@2022-09-17%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-aarch64
[+]          ^m4@1.4.19%gcc@11.1.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-aarch64
[+]              ^libsigsegv@2.14%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^perl@5.36.0%gcc@11.1.0+cpanm+open+shared+threads build_system=generic arch=linux-ubuntu20.04-aarch64
[+]              ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-aarch64
[+]      ^cmake@3.26.3%gcc@11.1.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-ubuntu20.04-aarch64
[+]          ^openssl@1.1.1t%gcc@11.1.0~docs~shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-aarch64
[+]              ^ca-certificates-mozilla@2023-01-10%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-aarch64
[+]      ^flex@2.6.3%gcc@11.1.0+lex~nls build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^findutils@4.9.0%gcc@11.1.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-aarch64
[+]      ^gmake@4.4.1%gcc@11.1.0~guile build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]      ^llvm@14.0.6%gcc@11.1.0+clang+compiler-rt~cuda~flang+gold+internal_unwind~ipo+libcxx+libomptarget~link_llvm_dylib+lld+lldb+llvm_dylib~mlir+omp_as_runtime~omp_debug~omp_tsan+polly~python~split_dwarf~z3 build_system=cmake build_type=Release generator=ninja patches=6379168,d85ef51,f920173 shlib_symbol_version=none targets=all version_suffix=none arch=linux-ubuntu20.04-aarch64
[+]          ^binutils@2.40%gcc@11.1.0~gas+gold~gprofng+headers~interwork+ld+libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-ubuntu20.04-aarch64
[+]              ^texinfo@7.0.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]              ^zstd@1.5.5%gcc@11.1.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-aarch64
[+]          ^hwloc@2.9.1%gcc@11.1.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-ubuntu20.04-aarch64
[+]              ^libpciaccess@0.17%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]                  ^libtool@2.4.7%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]                  ^util-macros@1.19.3%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]              ^libxml2@2.10.3%gcc@11.1.0~python build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^libedit@3.1-20210216%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^libffi@3.4.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^ninja@1.11.1%gcc@11.1.0+re2c build_system=generic arch=linux-ubuntu20.04-aarch64
[+]              ^re2c@2.2%gcc@11.1.0 build_system=generic arch=linux-ubuntu20.04-aarch64
[+]          ^perl-data-dumper@2.173%gcc@11.1.0 build_system=perl arch=linux-ubuntu20.04-aarch64
[+]          ^pkgconf@1.9.5%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]      ^ncurses@6.3%gcc@11.1.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]      ^python@3.7.15%gcc@11.1.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,f2fd060 arch=linux-ubuntu20.04-aarch64
[+]          ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-aarch64
[+]          ^expat@2.5.0%gcc@11.1.0+libbsd build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]              ^libbsd@0.11.7%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]                  ^libmd@1.0.4%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^gdbm@1.23%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^gettext@0.21.1%gcc@11.1.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]              ^tar@1.34%gcc@11.1.0 build_system=autotools zip=pigz arch=linux-ubuntu20.04-aarch64
[+]                  ^pigz@2.7%gcc@11.1.0 build_system=makefile arch=linux-ubuntu20.04-aarch64
[+]          ^libxcrypt@4.4.33%gcc@11.1.0~obsolete_api build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^readline@8.2%gcc@11.1.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-aarch64
[+]          ^sqlite@3.40.1%gcc@11.1.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^util-linux-uuid@2.38.1%gcc@11.1.0 build_system=autotools arch=linux-ubuntu20.04-aarch64
[+]          ^xz@5.4.1%gcc@11.1.0+pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-aarch64
[+]      ^zlib@1.2.13%gcc@11.1.0+optimize+pic+shared build_system=makefile arch=linux-ubuntu20.04-aarch64
</pre></details>

Error:
```
==> Installing ispc-1.19.0-vuhjrqb2g6mgca7tlsvfwueydaq6k5db
==> No binary for ispc-1.19.0-vuhjrqb2g6mgca7tlsvfwueydaq6k5db found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/c1/c1aeae4bdfb28004a6949394ea1b3daa3fdf12f646e17fcc0614861077dc8b6a.tar.gz
==> Ran patch() for ispc
==> ispc: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/cmake-3.26.3-7fcfio7un2wrsto7dn7mj4tof7jkea44/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/ispc-1.19.0-vuhjrqb2g6mgca7tlsvfwueydaq6k5db' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/ispc-1.19.0-vuhjrqb2g6mgca7tlsvfwueydaq6k5db/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/ispc-1.19.0-vuhjrqb2g6mgca7tlsvfwueydaq6k5db/lib64;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/llvm-14.0.6-cytur4nloibfhmbbexem5tnip72inzzw/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/binutils-2.40-ko6ybwpx256stcdjuiqey6eppustupbz/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/zlib-1.2.13-yzjmn5esi27hzi7hl5e5m3c3sfqrrae6/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/zstd-1.5.5-dzvxsh75mqmtldnaznhbgd4pl34ndryl/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/hwloc-2.9.1-nwydv6uo63nas5htscwv3uvlv6btg6k6/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libpciaccess-0.17-esijkqcln6ous62g32ydvrxf4uoj5nhl/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libxml2-2.10.3-bdu2fhhof6o6aqs7pfpq334662524mja/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libiconv-1.17-66xwheoww42j77iyf7rn6uhp7ltec3rz/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/xz-5.4.1-gigqtazkcwopt4uck24suhw2wv6apshe/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/ncurses-6.3-sp476v5v2p6ydz5h64lnhbnidhri3qfy/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libedit-3.1-20210216-anebouuinxelpxnlo44pu2vo4mpskjlm/lib;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libffi-3.4.4-775lb2t7itqbjblkoe2mlw56skpbonhq/lib' '-DCMAKE_PREFIX_PATH:STRING=<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/llvm-14.0.6-cytur4nloibfhmbbexem5tnip72inzzw;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/python-3.7.15-hmgvc54wc25hpr3acikxufxeuoyv56ba;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libffi-3.4.4-775lb2t7itqbjblkoe2mlw56skpbonhq;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libedit-3.1-20210216-anebouuinxelpxnlo44pu2vo4mpskjlm;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/hwloc-2.9.1-nwydv6uo63nas5htscwv3uvlv6btg6k6;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libpciaccess-0.17-esijkqcln6ous62g32ydvrxf4uoj5nhl;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/binutils-2.40-ko6ybwpx256stcdjuiqey6eppustupbz;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/zstd-1.5.5-dzvxsh75mqmtldnaznhbgd4pl34ndryl;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libxml2-2.10.3-bdu2fhhof6o6aqs7pfpq334662524mja;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/xz-5.4.1-gigqtazkcwopt4uck24suhw2wv6apshe;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/gmake-4.4.1-vrtzue4kj63kpno5orvem33e63lnzo7f;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/flex-2.6.3-pgpoltbiz22lbzbk5wuseddd76e7fnkr;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/cmake-3.26.3-7fcfio7un2wrsto7dn7mj4tof7jkea44;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/bison-3.8.2-qkerqrnbu2fu67ktaeyeivyiojx7iquy;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/zlib-1.2.13-yzjmn5esi27hzi7hl5e5m3c3sfqrrae6;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/ncurses-6.3-sp476v5v2p6ydz5h64lnhbnidhri3qfy;<truncated>/linux-ubuntu20.04-aarch64/gcc-11.1.0/libiconv-1.17-66xwheoww42j77iyf7rn6uhp7ltec3rz' '-DARM_ENABLED=FALSE' '-DISPC_NO_DUMPS=ON' '-DISPC_INCLUDE_EXAMPLES=OFF' '-DISPC_INCLUDE_TESTS=OFF' '-DISPC_INCLUDE_UTILS=OFF' '/tmp/root/spack-stage/spack-stage-ispc-1.19.0-vuhjrqb2g6mgca7tlsvfwueydaq6k5db/spack-src'

1 error found in build log:
     26    -- GIT_BINARY: /usr/bin/git
     27    fatal: not a git repository (or any of the parent directories): .git
     28    -- M4 macro processor: /spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde
           r__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-ubuntu20.04-aarch64/gcc-11.1.0/m4-1.4.19-cj3rou6nq77ndaw
           idf3bwovz3qkn5uri/bin/m4
     29    -- Found Python3: /usr/bin/python3.8 (found version "3.8.10") found components: Interpreter
     30    -- Found BISON: /spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__s
           pack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-ubuntu20.04-aarch64/gcc-11.1.0/bison-3.8.2-qkerqrnbu2fu67ktaeye
           ivyiojx7iquy/bin/bison (found suitable version "3.8.2", minimum required is "3.0")
     31    -- Found FLEX: /spack/opt/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__sp
           ack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholde/linux-ubuntu20.04-aarch64/gcc-11.1.0/flex-2.6.3-pgpoltbiz22lbzbk5wused
           dd76e7fnkr/bin/flex (found suitable version "2.6.3", minimum required is "2.6")
  >> 32    CMake Error at CMakeLists.txt:268 (message):
     33      Either X86, ARM, WASM or XE targets need to be enabled.
     34
     35
     36    -- Configuring incomplete, errors occurred!

See build log for details:
  /tmp/root/spack-stage/spack-stage-ispc-1.19.0-vuhjrqb2g6mgca7tlsvfwueydaq6k5db/spack-build-out.txt
```

### Error message

<details><summary>Error message</summary>

<pre>
... see above
</pre></details>

### Information on your system

* **Spack:** 0.20.0.dev0 (0dc6c421561a59e2b00b1daf634c1216f843daa0)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-neoverse_n1
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/11625789/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/11625809/spack-build-env.txt)

@aumuell @kwryankrattiger @svenevs @wspear @nchaimov

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by aumuell** [comment]
This is now reportedly fixed: https://github.com/spack/spack/pull/38080#issuecomment-1623895791 - should we close this?

**Comment by eugeneswalker** [comment]
> This is now reportedly fixed: [#38080 (comment)](https://github.com/spack/spack/pull/38080#issuecomment-1623895791) - should we close this?

Absolutely! Thanks!
